content/blog/learn-tidyverse-ai/index.Rmd

```diff
@@ -0,0 +1,542 @@
+Similarly, using `theme_minimal()` is not a bad or wrong choice either[^2], but it's not *necessary*, but this might not be obvious to a new learner.
+
+[^2]: In fact, it's my preferred ggplot2 theme!
+
+Furthermore, while ChatGPT "solves" the problem, a thorough code review reveals a number of not-so-great things about the answer that can be confusing for new learners or promote poor coding practices:
+
+-   The code loads packages that are not necessary: tidyr and ggplot2 packages are sufficient for this code, we don't also need dplyr. Additionally, learners coming from R for Data Science likely expect `library(tidyverse)` in analysis code, instead of loading the packages individualy.
+-   There is no need to load the `billboard` dataset, it's available to use once the tidyr package is loaded. Additionally, quotes are not needed, `data(billboard)` also works.
+-   The code mixes up tidyverse and base R styles:
+    -   Changing the type of `week` to numeric can be done in a `mutate()` statement with the tidyverse, which would then warrant loading the dplyr package.
+    -   This can also be done within `pivot_longer()` with the `names_transform` argument.
+
+All of these are addressable with further prompts, as I've done at [the saved chat](https://chatgpt.com/share/678f1dee-4af0-8000-befe-6df0a8d70fba), in the last two prompts.
+But doing so requires being able to identify these issues and explicitly asking for corrections.
+In practice, I wouldn't have asked ChatGPT to correct everything for me, I would have stopped after the first suggestion, which was a pretty good starting point, and made the improvements myself.
+However, a new learner might assume (and based on my experience seeing lots of new learner code, *does* assume) the first answer is the *right* and *good* or *best* answer since (1) it looks reasonable and (2) it works, sort of.
+
+Furthermore, requesting improvements in subsequent calls can result in surprising changes that the user hasn't asked for.
+We saw an example of this above, in updating the alpha level.
+Similarly, in [the saved chat](https://chatgpt.com/share/678f1dee-4af0-8000-befe-6df0a8d70fba) you can see that asking ChatGPT to not load the packages individually but to use `library(tidyverse)` instead results in this change as well as not loading the data with a `data()` call and adding a data transformation step with `mutate()` to convert `week` to numeric:
+
+```{r}
+#| message: false
+# Load the tidyverse package
+library(tidyverse)
+
+# Load the billboard dataset and prepare the data
+billboard_long <- billboard |>
+  pivot_longer(
+    cols = starts_with("wk"),
+    names_to = "week",
+    values_to = "rank",
+    names_prefix = "wk",
+    values_drop_na = TRUE
+  ) |>
+  mutate(week = as.numeric(week))  # Convert week to numeric
+```
+
+Both of these are welcome changes, but it can be surprising to a new learner why they're combined with updating the `library()` call.
```

Comment:
I think it's worth a comment somewhere that you generally get better response when asking questions about sample datasets because the model has probably been exposed to them somewhere in their training data.

Reply:
Added, thanks for the suggestion